test/e2e/prune_test.go

```diff
@@ -580,4 +588,69 @@ var _ = Describe("Podman prune", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(dirents).To(HaveLen(3))
 	})
+
+	It("podman system prune --build clean up after terminated build", func() {
+		useCustomNetworkDir(podmanTest, tempdir)
+		containerStorageDir := filepath.Join(podmanTest.Root, podmanTest.ImageCacheFS+"-containers")
+
+		podmanTest.BuildImage(pruneImage, "alpine_notleaker:latest", "false")
+
+		create := podmanTest.Podman([]string{"create", "--name", "test", BB})
+		create.WaitWithDefaultTimeout()
+		Expect(create).Should(ExitCleanly())
+
+		containerFilePath := filepath.Join(podmanTest.TempDir, "ContainerFile-"+stringid.GenerateRandomID())
+		err := os.WriteFile(containerFilePath, []byte(longBuildImage), 0755)
+		Expect(err).ToNot(HaveOccurred())
+
+		build := podmanTest.Podman([]string{"build", "-f", containerFilePath, "-t", "podmanleaker"})
+		// Build will never finish so let's kill it after 8 seconds to simulate a failed build that leaves stage containers
+		time.Sleep(8 * time.Second)
```

Comment:
Why do we wait 8s? That slows down the test a lot, 1/2 s should be more than enough?!